packages/seal-mongodb-adapter/src/ClientWrapper.php

```diff
@@ -0,0 +1,36 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of the CMS-IG SEAL project.
+ *
+ * (c) Alexander Schranz <alexander@sulu.io>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace CmsIg\Seal\Adapter\MongoDB;
+
+use MongoDB\Client;
+use MongoDB\Database;
+
+final class ClientWrapper
+{
+    public function __construct(
+        private readonly Client $client,
+        private readonly string $databaseName,
+    ) {
+    }
+
+    public function getClient(): Client
+    {
+        return $this->client;
+    }
+
+    public function getDatabase(): Database
+    {
+        return $this->client->getDatabase($this->databaseName);
+    }
+}
```

Comment:
Not sure about this wrapper class didn't want that the services always require:

```
        private readonly Client $client,
        private readonly string $databaseName,
```

I'm also not sure if my DSN will conflict with MongoDB URI parsing I'm using the DSN `path` part to select the database:

https://github.com/PHP-CMSIG/search/blob/903fe7414b6a168dc7f4758e4a16038dfae13012/packages/seal-mongodb-adapter/src/MongoDBAdapterFactory.php#L75-L85

Reply:
The default database can be specified in the "path" part of the DSN: https://www.mongodb.com/docs/manual/reference/connection-string/#connection-string-database-options

The `MongoDB\Client` class doesn't have the notion of default database, you have to extract the default database name from the DSN (or other options).

Reply:
I would try to avoid extracting the database name from the connection string - it would require you to parse the connection string and thus be conformant with the above specification. The default database was used by some tools, but the MongoDB drivers don't leverage it for anything other than authentication.